tests/components/mill/test_init.py

```diff
@@ -42,9 +39,10 @@ async def test_setup_with_cloud_config_fails(hass):
         },
     )
     mock_entry.add_to_hass(hass)
-    with patch("mill.Mill.connect", return_value=False):
-        with pytest.raises(ConfigEntryNotReady):
-            assert not await mill.async_setup_entry(hass, mock_entry)
+    with patch("mill.Mill.connect", return_value=False), pytest.raises(
+        ConfigEntryNotReady
+    ):
+        assert not await mill.async_setup_entry(hass, mock_entry)
```

Comment:
Sorry I missed this earlier: I also realized that this is another example where there is a preferred API to set up the component, rather than accessing the integration methods directly: https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

I think the most common way I see this is to just invoke `async_setup_component` which should also invoke your `async_setup_entry` method with the config entry created a few lines above.

(Here and in a few other places)